src/view/views/MyProgramsView.tsx

```diff
@@ -35,9 +35,20 @@ export const MyProgramsView = () => {
       dispatch(
         blocklySlice.actions.setActiveBlocklyProgramId({ title: program.title })
       );
+
+      for (const entry of programs) {
```

Comment:
i guess the intention is to search for the matching program. but there could be duplicates. if this really is a list then we likely want to match the first instance. if its actually an object and each program a property then we can just programs[entry.title] or similar (i havent looked)

Reply:
> i guess the intention is to search for the matching program.

Yes, the intention is to search for the matching program.   We now match the first instance, as suggested.

> but there could be duplicates

I believe there should be no duplicates.  Because the program title is used as the identifier for the program, so each program title has to be unique.  I think our codebase ensure that this invariant hold.  E.g. if you save the current code and give it a title that is already taken then you'll save over the existing program, as opposed to creating a second program with the same title.

I guess the underlying issue here is that we store the programs in an array, instead of a map from program id (title) to program objects (description and xml-encoded blockly code).  That way we could just do map look-ups instead of looping through our array.